### ISSUE TITLE
Update  Go versions used in CI

### ISSUE THREAD

**Issue by garyburd** [gorilla/websocket#792]
- Remove < 1.13 and possibly < 1.16
- Add 1.18

The current CircleCi config fails when Go 1.18 is added.

See related
-  #788 (requires Go >= 1.13)
-  #776 (requires Go >= 1.16)
- #778 (edits CircleCI config, adds GitHub action)

I do not have time to work on this. It will be great if somebody else can investigate.

**Comment by garyburd** [comment]
We can move from Circle Ci to something else.

**Comment by ChannyClaus** [comment]
Oh oops, I just got the PR ready for a review where I added 1.18 for Circle CI - can prepare another that moves the CI off Circle CI if that's preferred.

**Comment by amustaque97** [comment]
> Oh oops, I just got the PR ready for a review where I added 1.18 for Circle CI - can prepare another that moves the CI off Circle CI if that's preferred.

Initially, I was in favor of moving from Circle CI to Github Actions. I went through a couple of blogs and links. I think we are doing okay with Circle CI. 

Here are the links to articles I went through:
1. https://medium.com/instrument-stories/using-github-actions-and-circleci-b0a30c0a2608
2. https://coderonfleek.medium.com/circleci-vs-github-actions-a-software-engineers-perspective-14567e539b9c
3. https://circleci.com/circleci-versus-github-actions/
4. https://www.tinystacks.com/blog-post/circleci-vs-github-actions-which-is-better/


I will review the PR in some time. 

Thank you
